extensions/ima/src/main/java/com/google/android/exoplayer2/ext/ima/AdTagLoader.java

```diff
@@ -1122,9 +1122,15 @@ private void handleAdPrepareError(int adGroupIndex, int adIndexInAdGroup, Except
   }
 
   private void ensureSentContentCompleteIfAtEndOfStream() {
+    if (player == null) {
+      return;
+    }
+
+    int pendingAdGroupIndex = adPlaybackState.getAdGroupIndexAfterPositionUs(player.getContentPosition(), player.getContentDuration());
```

Comment:
Instead of player.getContentPosition and player.getContentDuration, you need to use the values already used below (`getContentPeriodPositionMs(...)` and `contentDurationMs`).  

To make this easier, maybe reorder the method slightly:
 1. Check for the simple conditions first and return from this method if they don't apply. 
 2. Get the content position using the helper method `getContentPeriodPositionMs`.
 3. Check the more complicated conditions to see if we need to trigger `sendContentComplete`.